[Validations] Validations fail when using dimension: { min: 1080..1080 } syntax

Mth0158 opened the pull request:
Closes #163 

This PR has 2 commits :
- [[Bugfix] Correct test fail using 'dimension: { min: 1080..1080 }'](https://github.com/igorkasyanchuk/active_storage_validations/commit/8da49577bd9bcf8d7d0b23699f9e38065a65f33c) => fixes #163 
- [[DimensionValidator] Make semantic changes for better understanding](https://github.com/igorkasyanchuk/active_storage_validations/commit/810f8eb8f0a4f405fef263b9200d394c41f7022f) => makes semantic changes for better understanding

The second commit is optional, though I think it's a good thing to change some methods & calls for making contributing slightly easier

Open to discussion, I would like to contribute further to this amazing project that you made @igorkasyanchuk ;)

gr8bit commented:
@Mth0158 if you synced with the current master the minutest problems should be gone. :)

Mth0158 commented:
@gr8bit all good, I have rebased onto `master` with your changes, all tests pass now.

I have also added the `dimension: { max: xxx ... yyy }` part since the problem would be the same as `min`. Let me know if the 2 commits are okay.

I can also help on other subjects if needed!

gr8bit commented on test/matchers/dimension_validator_matcher_test.rb:
```diff
@@ -143,4 +143,62 @@ class ActiveStorageValidations::Matchers::DimensionValidatorMatcher::Test < Acti
+    matcher = ActiveStorageValidations::Matchers::DimensionValidatorMatcher.new(:dimension_min)
+    matcher.width_min 700
+    matcher.height_min 600
+    refute matcher.matches?(Project)
```
Just because I saw it above - shouldn't this be `Project.new`?

Mth0158 replied:
Well, I have followed the same test pattern as the other tests. Here we are testing the test matcher, if we pass a class (`Project`) to the `matches?` method, it will convert it to `Project.new` automatically in the method.

gr8bit commented on test/matchers/dimension_validator_matcher_test.rb:
```diff
@@ -143,4 +143,62 @@ class ActiveStorageValidations::Matchers::DimensionValidatorMatcher::Test < Acti
     matcher.width_min 800
     refute matcher.matches?(Project.new)
```
Regarding my comment below, I was referring to this line. But you're right and we should probably pass the class instead of an object everywhere.

gr8bit commented:
@Mth0158 thank you for your contribution! 👍
